check_labels.R

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env Rscript
+
+# Remove non-aplhanumeric characters 
+# set to lower-case, if scpecified
+
+suppressPackageStartupMessages(require(optparse))
+suppressPackageStartupMessages(require(workflowscriptscommon))
+suppressPackageStartupMessages(library(reshape2))
+suppressPackageStartupMessages(library(data.table))
+
+option_list = list(
+    make_option(
+        c("-i", "--input-file"),
+        action = "store",
+        default = NA,
+        type = 'character',
+        help = 'Path to input metadata file in .tsv format'
+    ),
+    make_option(
+        c("-l", "--label-field"),
+        action = "store",
+        default = NA,
+        type = 'character',
+        help = 'Name of label field in metadata file'
+    ),
+    make_option(
+        c("-k", "--condensed"),
+        action = "store_true",
+        default = FALSE,
+        type = 'logical',
+        help = 'Is the provided metadata file in condensed format? Default: False'
+    ),
+    make_option(
+        c("-a", "--avoid-lowercase"),
+        action = "store_true",
+        default = FALSE,
+        type = 'logical',
+        help = 'Should setting the labels to lowercase be skipped? Default: False'
+    ),
+    make_option(
+        c("-o", "--output-path"),
+        action = "store",
+        default = NA,
+        type = 'character',
+        help = 'Output for updated file'
+    )
+)
+
+# parse arguments 
+opt = wsc_parse_args(option_list, mandatory = c("input_file", "label_field", "output_path"))
+# source function definitions 
+script_dir = dirname(strsplit(commandArgs()[grep('--file=', commandArgs())], '=')[[1]][2])
+source(file.path(script_dir, 'cell_types_utils.R'))
+
+if(!file.exists(opt$input_file)){
+    stop("Input file is not provided")
+}
+
+reg_exp = "[^-A-Za-z0-9>+ ]"
+cond = opt$condensed
+lab_field = opt$label_field
+
+if(cond){
+    data = fread(opt$input_file, header=FALSE, stringsAsFactors = FALSE, fill = TRUE)
+    # extract rows with labels 
+    idx = which(data[, 5] == lab_field)
```

Comment:
The number of fields in a condensed SDRF is irritatingly variable, so you need to parameterise the condensed SDRF field numbers (the values you have here and below can be your sensible defaults). See https://github.com/ebi-gene-expression-group/experiment_metadata/blob/develop/unmelt_condensed.R for my previous notes on this.